### DISCUSSION TITLE
API Call to listNamespacesWithPattern Method in org.apache.kyuubi.engine.spark.util.SparkCatalogUtils.getCatalogTablesOrViews

### DISCUSSION THREAD

**Discussion by mahalingamagesthian** [apache/kyuubi#6059] [General]
Background. We are using the Kyuubi JDBC driver to connect to a prominent BI Tool. During this,
We have identified that the API call to the method `listNamespacesWithPattern`, from `org.apache.kyuubi.engine.spark.util.SparkCatalogUtils.getCatalogTablesOrViews` may not be required. 
Details:  The following operation, 
`getCatalogTablesOrViews(spark,"spark_catalog","dbname","*",tableType,true)`
or 
`get CatalogTablesOrViews(spark,"spark_catalog","dbname","tablename",tableType,true)`
invokes
```
val namespaces = listNamespacesWithPattern(catalog,"dbname")
namespaces: Array[Array[String]] = Array(Array(dbname))
```

This Goes for a redundant listing of all the databases in Hive, it is happening at at https://github.com/apache/kyuubi/blob/branch-1.8/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/util/SparkCatalogUtils.scala#L163 .In order to perform this operation, it took about a minute.
The code block: 
https://github.com/apache/kyuubi/blob/branch-1.8/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/util/SparkCatalogUtils.scala#L163
is invoked, there is no opportunity to use the namespaces further.
The code block: starting from 
https://github.com/apache/kyuubi/blob/branch-1.8/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/util/SparkCatalogUtils.scala#L209
untill
https://github.com/apache/kyuubi/blob/branch-1.8/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/util/SparkCatalogUtils.scala#L223
may not be required as well.

Please let me know what are your thoughts on this, Thanks.

**Comment by pan3793** [comment]
sorry for the late reply, just back to the office from a long holiday.

instead of talking about code details in a text conversation, would you mind sending a PR directly?

**Reply by mahalingamagesthian** [reply]
I am new to doing the PRs. Is there a set of guidelines that I can read through, learn and raise this as PR ?

**Reply by pan3793** [reply]
We have a Code Contribution Program before, and you may find useful guidance at https://github.com/apache/kyuubi/issues/5357
